linksharing/handler.go

```diff
@@ -307,3 +336,138 @@ func makeLocation(base *url.URL, reqPath string) string {
 	location.Path = path.Join(location.Path, reqPath)
 	return location.String()
 }
+
+// handleHostingService deals with linksharing via custom URLs.
+func (handler *Handler) handleHostingService(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	host := strings.SplitN(r.Host, ":", 2) // This handles the case where the host contains a port number.
```

Comment:
you may want to use net.SplitHostPort in case this is an ipv6 address (which has colons)

Reply:
ah nice will do

Reply:
done